interfaces/MatlabToolbox/matlab_new/Thermo/ThermoPhase.m

```diff
@@ -0,0 +1,1736 @@
+            %     Instance of class :mat:func:`ThermoPhase` (or another
+            %     object that derives from ThermoPhase)
+            % :return:
+            %     Number of elements in the phase.
+            %
+            nel = calllib(ct, 'thermo_nElements', tp.tpID);
+        end
+        
+        function nsp = nSpecies(tp)
+            % NSPECIES  Get the number of species.
+            % n = tp.nSpecies
+            %
+            % :param tp:
+            %     Instance of class :mat:func:`ThermoPhase` (or another
+            %     object that derives from ThermoPhase)
+            % :return:
+            %     Number of species in the phase.
+            %
+            nsp = calllib(ct, 'thermo_nSpecies', tp.tpID);
+        end
+        
+        function k = speciesIndex(tp, name)
+            % SPECIESINDEX  Get the index of a species given the name.
+            % k = tp.speciesIndex(name)
+            % The index is an integer assigned to each species in sequence as it
+            % is read in from the input file.
+            %
+            % NOTE: In keeping with the conventions used by Matlab, this method
+            % returns 1 for the first species, 2 for the second, etc. In
+            % contrast, the corresponding method speciesIndex in the Cantera C++
+            % and Python interfaces returns 0 for the first species, 1 for the
+            % second one, etc. ::
+            %
+            %     >> ich4 = gas.speciesIndex('CH4');
+            %     >> iho2 = gas.speciesIndex('HO2');
+            %
+            % :param tp:
+            %     Instance of class :mat:func:`ThermoPhase` (or another
+            %     class derived from ThermoPhase)
+            % :param name:
+            %     If name is a single string, the return value will be a integer
+            %     containing the corresponding index. If it is an cell array of
+            %     strings, the output will be an array of the same shape
+            %     containing the indices.
+            % :return:
+            %     Scalar or array of integers
+            %
+            if iscell(name)
+                [m, n] = size(name);
+                k = zeros(m, n);
+                for i = 1:m
+                    for j = 1:n
+                        k(i, j) = calllib(ct, 'thermo_speciesIndex', ...
+                                          tp.tpID, name{i, j}) + 1;
+                        if k(i, j) > 1e3
+                            warning(['Species ', name{i, j}, ...
+                                   ' does not exist in the phase']);
+                            k(i, j) = -1;
+                        end
+                    end
+                end
+            elseif ischar(name)
+                k = calllib(ct, 'thermo_speciesIndex', ...
+                            tp.tpID, name) + 1;
+                if k > 1e3
+                    warning(['Species ', name, ...
+                           ' does not exist in the phase.']);
```

Comment:
I promise there are input files defining phases with more than 1000 species.

Reply:
I set the new limit to 1e6, would that be a safe value?

Reply:
Is there a reason to handle this case in Matlab, instead of relying on exception handling that exists in C++?

Reply:
I think the issue is struggling with the fact that this method returns `npos` if the species doesn't exist -- it doesn't raise an exception.

Reply:
Something like this is possible
```matlab
                k = uint64(calllib(ct, 'thermo_speciesIndex', tp.tpID, name)) + 1;
                if k == intmax('uint64')
                    warning(['Species ', name, ' does not exist in the phase.']);
                    k = uint64(0);
```
The Note at [this matlab documantion page](https://www.mathworks.com/help/matlab/matlab_external/passing-arguments-to-shared-library-functions.html) points out "All scalar values returned by MATLAB are of type double." 
Which got me wondering is there value in us converting some of them to other types. (Like sometimes some form of integer, be it uint64 or another).

Reply:
(`uint32` may be better, because math operations done with `uint64` take ultra precision before being converted back, which might be more costly)